### PR TITLE
"Not" Scan Type Feature

### PR DESCRIPTION
**Pull request by OpenNetSurfer**
- A new option in scan type `comboBox` is added
- `tr.py` is updated to include string `Not`
- Option works only with `INT` and `FLOAT` value type

### CODE REVIEW

File: PINCE.py
Comment (brkzlr) [inline PINCE.py:1110]: This logic should be in `validate_search`. It looks ugly being the only one outside, plus if validate_search is ever used elsewhere, the Not functionality will be broken.

File: PINCE.py
Comment (brkzlr) [inline PINCE.py:1259]: I don't think deletion is the best approach here. Try looking at how `INCREASED`, `DECREASED` and so on are hidden from the user in code and emulate that behaviour.

**Review by brkzlr — CHANGES_REQUESTED** [review]

File: PINCE.py
Comment (OpenNetSurfer) [reply PINCE.py:1110]: I see, will try to put it there.

File: PINCE.py
Comment (OpenNetSurfer) [reply PINCE.py:1259]: Are you mentioning the `get_list()` function?

From what I can tell `pushButton_NewFirstScan_clicked()` always calls `comboBox_ScanType_init()`. From here `get_list(self.scan_mode)` is called which will return `INCREASED`, `DECREASED` if scan_mode is `ONGOING`.

So, I should modify the `get_list()` function?

Comment (brkzlr) [reply PINCE.py:1259]: Yes, you should pass the value type to that function as well so you can check inside and see whether `NOT` should be in or not.

This way it's more portable and we don't do "hacky" deletions in random places in code.

Comment (OpenNetSurfer) [reply PINCE.py:1259]: Is this better?

```python
    def get_list(scan_mode, value_type):
        list = []
        if scan_mode == SCAN_MODE.NEW:
            list = [
                SCAN_TYPE.EXACT,
                SCAN_TYPE.NOT,
                SCAN_TYPE.LESS,
                SCAN_TYPE.MORE,
                SCAN_TYPE.BETWEEN,
                SCAN_TYPE.UNKNOWN,
            ]
        else:
            list = [
                SCAN_TYPE.EXACT,
                SCAN_TYPE.NOT,
                SCAN_TYPE.INCREASED,
                SCAN_TYPE.INCREASED_BY,
                SCAN_TYPE.DECREASED,
                SCAN_TYPE.DECREASED_BY,
                SCAN_TYPE.LESS,
                SCAN_TYPE.MORE,
                SCAN_TYPE.BETWEEN,
                SCAN_TYPE.CHANGED,
                SCAN_TYPE.UNCHANGED,
            ]

        # Delete Not option
        if value_type == SCAN_INDEX.AOB or value_type == SCAN_INDEX.STRING:
            del list[1]
        
        return list
```

Comment (brkzlr) [reply PINCE.py:1259]: Yes, get rid of the comment, it's useless.

Comment (OpenNetSurfer) [reply PINCE.py:1259]: Ok, thanks!

File: libpince/typedefs.py
Comment (brkzlr) [inline libpince/typedefs.py:335]: Just FYI, this line is not needed since there is no block scope in Python. Anything declared in the if and else below belong to the whole function.

No need to make another commit to change this though, just have it in mind for the future.

**Review by brkzlr — APPROVED** [review]
Seems fine so far.

One thing to note is to please refrain from committing stuff with "(Ignore)" in title and such in the future. No commit should be ignored.

I'll let Korcan take a look/test as well before we merge this.

File: libpince/typedefs.py
Comment (korcankaraokcu) [reply libpince/typedefs.py:335]: It's aight, I can delete the redundant code later on

**Review by korcankaraokcu — APPROVED** [review]

**Comment by korcankaraokcu** [comment]
Thanks for the patch!

**Comment by OpenNetSurfer** [comment]
> One thing to note is to please refrain from committing stuff with "(Ignore)" in title and such in the future. No commit should be ignored.

I understand. Will not do that in future

And I am glad I could contribute!

**Pull request merged**
